bundles/org.openhab.binding.elgatokeylight/src/main/java/org/openhab/binding/elgatokeylight/internal/ElgatoKeyLight.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.elgatokeylight.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+
+/**
+ * Simple tool for talking to the Elgato Key Light API as documented <a href=
+ * "https://github.com/bitfocus/companion-module-requests/issues/198">here</a>.
+ *
+ * @author Gunnar Wagenknecht - Initial contribution
+ */
+public class ElgatoKeyLight {
+
+    public static class LightStatus {
+
+        public boolean on;
+        public int brightness;
+        public int temperature;
+    }
+
+    private static final String TEMPERATURE = "temperature";
+    private static final String BRIGHTNESS = "brightness";
+    private static final String ON = "on";
+
+    private static final String DISPLAY_NAME = "displayName";
+    private static final String SERIAL_NUMBER = "serialNumber";
+    private static final String PRODUCT_NAME = "productName";
+
+    private final String ip;
+    private final JsonParser parser;
+    private final Gson gson;
+    private volatile JsonObject accessoryInfo;
+    private final int port;
+
+    public ElgatoKeyLight(final String ip, final int port) {
+        this.ip = ip;
+        this.port = port;
+        parser = new JsonParser();
+        gson = new GsonBuilder().create();
+    }
+
+    private JsonObject createLightStatus(final Boolean on, final Integer brightness, final Integer temperature) {
+        JsonObject desiredLightState = new JsonObject();
+        desiredLightState.add("numberOfLights", new JsonPrimitive(1));
+        JsonArray lights = new JsonArray(1);
+        desiredLightState.add("lights", lights);
+        JsonObject light = new JsonObject();
+        lights.add(light);
+        if (on != null) {
+            light.addProperty(ON, on ? 1 : 0);
+        }
+        if (brightness != null) {
+            light.addProperty(BRIGHTNESS, brightness);
+        }
+        if (temperature != null) {
+            light.addProperty(TEMPERATURE, temperature);
+        }
+        return desiredLightState;
+    }
+
+    private JsonObject fetchAccessoryInfo() throws IOException {
+        String accessoryInfoResponse = HttpUtil.executeUrl(HttpMethod.GET.asString(), getUrl("/elgato/accessory-info"),
+                30);
```

Comment:
A timeout of 30 sec. seems a bit long. This would block an OH scheduler thread. 10 sec. would be a good value. Consider make this a static final constant.